src/common/gpio_motor.rs

```diff
@@ -591,10 +654,52 @@ where
     }
 
     fn stop(&mut self) -> anyhow::Result<()> {
+        set_enable_pins(
+            &mut self.board,
+            self.motor_settings.enable_pin_high,
+            self.motor_settings.enable_pin_low,
+            false,
+        )?;
         self.board.set_pwm_duty(self.pwm_pin, 0.0)?;
         self.board.set_gpio_pin_level(self.a_pin, false)?;
         self.board.set_gpio_pin_level(self.b_pin, false)?;
         self.is_on = false;
         Ok(())
     }
 }
+
+fn set_enable_pins(
+    board: &mut dyn Board,
+    enable_high_pin: Option<i32>,
+    enable_low_pin: Option<i32>,
+    enable: bool,
+) -> anyhow::Result<()> {
+    let enable_high_result = match enable_high_pin {
+        Some(pin) => board.set_gpio_pin_level(pin, enable),
+        None => Ok(()),
+    };
+    let enable_low_result = match enable_low_pin {
+        Some(pin) => board.set_gpio_pin_level(pin, !enable),
+        None => Ok(()),
+    };
+    match (enable_high_result, enable_low_result) {
+        (Err(err), Ok(_)) => Err(anyhow::anyhow!(
+            "motor could not set enable high pin to {:?} due to {:?}",
+            enable,
+            err
+        )),
+        (Ok(_), Err(err)) => Err(anyhow::anyhow!(
+            "motor could not set enable low pin to {:?} due to {:?}",
+            !enable,
+            err
+        )),
+        (Err(high_err), Err(low_err)) => Err(anyhow::anyhow!(
```

Comment:
I just don't understand the point off this double error. Why do we try to set a pin to if the first one defined failed?

Reply:
I think we should make this flow nicer, I don't think we have motor drivers that would require 2 enable pins unless it some sort of error clearing mechanism and in this case we should not be clearing it like this without the user making a more educated decision to reset the error.
For motor driver with power saving features We can leave the enable pin but it should only be called at the construction of the motor only.